transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -105,34 +116,40 @@ public final Channel channel() {
     }
 
     @Override
-    public final ChannelPipeline addFirst(String name, ChannelHandler handler) {
-        return addFirst(null, name, handler);
+    public final EventExecutor executor() {
+        return executor;
     }
 
     @Override
-    public final ChannelPipeline addFirst(EventExecutor executor, String name, ChannelHandler handler) {
+    public final ChannelPipeline addFirst(String name, ChannelHandler handler) {
         final AbstractChannelHandlerContext newCtx;
-        synchronized (this) {
-            checkMultiplicity(handler);
-            name = filterName(name, handler);
```

Comment:
Why do we no longer need to check for duplicate names?

Reply:
We do... it was just moved a bit

Reply:
I saw it, thanks.